### PR TITLE
Rename Reaper metrics to make them compatible with Prometheus

### PR DESCRIPTION
**Pull request by rzvoncek**

### CODE REVIEW

File: src/server/src/main/java/io/cassandrareaper/jmx/HostConnectionCounters.java [hunk 1]
```diff
@@ -45,9 +45,9 @@ void incrementSuccessfulConnections(String host) {
       LOG.debug("Host {} has {} successfull connections", host, successes);
       if (null != metricRegistry) {
         metricRegistry
-            .counter(
-                MetricRegistry.name(
-                    JmxConnectionFactory.class, "connections", host.replace('.', '-')))
+            .counter(MetricRegistry.name(JmxConnectionFactory.class,
+                                         "connections",
+                                         host.replace('.', 'x').replaceAll("[^A-Za-z0-9]", "")))
```
Comment (michaelsembwever) [inline src/server/src/main/java/io/cassandrareaper/jmx/HostConnectionCounters.java]: Can you undo the line wrapping changes. I believe the additional .replaceAll call doesn't break the line-length code style rule.
```suggestion
            .counter(
                MetricRegistry.name(
                    JmxConnectionFactory.class, "connections",host.replace('.', 'x').replaceAll("[^A-Za-z0-9]", "")))                                        
```

File: src/server/src/main/java/io/cassandrareaper/service/RepairRunner.java [hunk 2]
```diff
@@ -587,7 +587,8 @@ void killAndCleanupRunner() {
   private String metricName(String metric, String clusterName, String keyspaceName, UUID repairRunId) {
     String cleanClusterName = clusterName.replaceAll("[^A-Za-z0-9]", "");
     String cleanRepairRunId = repairRunId.toString().replaceAll("-", "");
-    return MetricRegistry.name(RepairRunner.class, metric, cleanClusterName, keyspaceName, cleanRepairRunId);
+    String cleanKeyspaceName = keyspaceName.replaceAll("[^A-Za-z0-9]", "");
```
Comment (michaelsembwever) [inline src/server/src/main/java/io/cassandrareaper/service/RepairRunner.java]: hasn't `cleanKeyspaceName` already replaced those characters?

**Comment by michaelsembwever** [comment]
Look good. But you'll need to fix the indentation.

And the tests are failing as SegmentRunnerTest doesn't set the keyspaceName in the RepairUnit it mocks. (SegmentRunnerTest should be fixed, as production code would never have a null keyspaceName.)

**Comment by rzvoncek** [comment]
Reverted the indentation and fixed the SegmentRunnerTest. Had to remove one blank line because checkstyle was complaining about the method being too long (151 vs 150 allowed lines).

I've also rebased on current master.

**Review by michaelsembwever — APPROVED** [review]
:shipit:

**Comment by rzvoncek** [comment]
Squashed.

**Comment by michaelsembwever** [comment]
Manually merged to `1.3` and `master`.

In `1.3` it is a compatibility break, but in a rather controlled and cosmetic way I think we can clearly communicate and explain.

**Pull request closed without merging**